web/markdown/rules.react.js

```diff
@@ -47,8 +71,105 @@ function linkRules(): MarkdownRuleSpec {
+    },
+    blockQuote: {
+      ...SimpleMarkdown.defaultRules.blockQuote,
+      // match end of blockQuote by either \n\n or end of string
+      match: SimpleMarkdown.blockRegex(/^( *>[^\n]+(\n[^\n]+)*\n*)+(\n{2,}|$)/),
+      parse(
+        capture: SimpleMarkdown.Capture,
+        parse: SimpleMarkdown.Parser,
+        state: SimpleMarkdown.State,
+      ) {
+        var content = capture[0].replace(/^ *> ?/gm, '');
+        return {
+          // remove new line from captured string
+          content: parse(content.trim(), state),
+        };
+      },
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => (
+        <blockquote key={state.key} className={css.block}>
+          {output(node.content, state)}
+        </blockquote>
+      ),
+    },
+    inlineCode: {
+      ...SimpleMarkdown.defaultRules.inlineCode,
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => (
+        <code key={state.key} className={css.inlineCode}>
+          {node.content}
+        </code>
+      ),
+    },
+    em: SimpleMarkdown.defaultRules.em,
+    strong: SimpleMarkdown.defaultRules.strong,
+    del: SimpleMarkdown.defaultRules.del,
+    underline: {
+      order: SimpleMarkdown.defaultRules.em.order - 0.5,
+      match: SimpleMarkdown.inlineRegex(/^__([\s\S]+?)__(?!_)/),
+      parse: (
+        capture: SimpleMarkdown.Capture,
+        parse: SimpleMarkdown.Parser,
+        state: SimpleMarkdown.state,
+      ) => ({
+        content: parse(capture[1], state),
+      }),
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => <u key={state.key}>{output(node.content, state)}</u>,
+    },
+    // heading: {
+    //   ...SimpleMarkdown.defaultRules.heading,
+    //   match: SimpleMarkdown.inlineRegex(/^ *(#{1,6} )([^\n]+?) *(?:\n+|$)/),
+    // },
+    mailto: SimpleMarkdown.defaultRules.mailto,
+    codeBlock: {
+      ...SimpleMarkdown.defaultRules.codeBlock,
+      match: SimpleMarkdown.blockRegex(/^(?: {4}[^\n]+\n*)+(?:\n *)/),
+      parse(capture: SimpleMarkdown.Capture) {
+        var content = capture[0].replace(/^ {4}/gm, '');
+        return {
+          lang: undefined,
+          content: content,
+        };
+      },
+    },
+    fence: {
+      ...SimpleMarkdown.defaultRules.fence,
+      match: SimpleMarkdown.blockRegex(/`{3}(?:(.*$)\n)?([\s\S]*)`{3}/m),
+      parse(capture: SimpleMarkdown.Capture) {
+        return {
+          type: 'codeBlock',
+          lang: capture[1] || undefined,
```

Comment:
By specifying `lang` here you're going to make `codeBlock`'s `react` method add a `className`. Since we don't actually have CSS classes for custom languages, let's avoid specifying this.

I think we can just use `parseBlock` if we make the `language` capture group in the RegEx a non-capture (`(?:)`) group.